Select the smaller image size from the list of available sources

jjgrainger opened the pull request:
## Summary

Fixes #186 Select the smaller image size from the list of available sources

## Relevant technical choices

## Checklist

- [ ] PR has either `[Focus]` or `Infrastructure` label.
- [ ] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

jjgrainger committed:
setup sources functions

jjgrainger committed:
calculate performant mime type for attachment

mitogh commented on modules/images/webp-uploads/load.php:
```diff
@@ -603,3 +603,78 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
+	}
+
+	// Return the resized image sources.
+	if ( isset( $metadata['sizes'][ $size ]['sources'] ) && ! empty( $metadata['sizes'][ $size ]['sources'] ) ) {
```
In this case `empty` works simiarly to `isset` so you can avoid the `isset` call.

```suggestion
	if ( ! empty( $metadata['sizes'][ $size ]['sources'] ) ) {
```

mitogh commented on modules/images/webp-uploads/load.php:
```diff
@@ -603,3 +603,78 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
+	}
+
+	// Return an empty array if no sources found.
+	return [];
```
We are using the long syntax array for arrays due to WP styleguide.

```suggestion
	return array();
```

jjgrainger commented:
Thanks @mitogh 

As you suggested I've created a new PR that branches from other work this expands on.

I've addressed the feedback in the new PR #243 

Closing this.
